ta/pkcs11/include/pkcs11_ta.h

```diff
@@ -469,11 +498,200 @@ struct pkcs11_mechanism_info {
 #define PKCS11_CKFM_EC_UNCOMPRESS		(1U << 24)
 #define PKCS11_CKFM_EC_COMPRESS			(1U << 25)
 
+/*
+ * pkcs11_object_head - Header of object whose data are serialized in memory
+ *
+ * An object is made of several attributes. Attributes are stored one next to
+ * the other with byte alignment as a serialized byte arrays. Appended
```

Comment:
Updating the text.

As far as I understand everything is 1 byte aligned so the marshalling routines has to handle that when accessing the structs.

> And C struct members are written with "." ("::" is C++).

I think neither is valid C syntax. A reference to the C spec would be welcome.

Reply:
To my understanding "prepending" refers to the struct as it is layed out in memory. Under normal circumstances this struct will be properly packed and there will be no padding between fields. (a good goto for this topic is http://www.catb.org/esr/structure-packing/).
This should not be an issue in this case, as there are only 32-bit ints (and the variable array at the end) in this struct. but something to keep in mind.

The other comment was mainly that to my experience "::" is very uncommon for C code documentation (it is used in C++ for namespacing) . Direct struct member access is done by "." and thus I prefer to keep this scheme also in the documentation _unless_ you are sticking to Doxygen, see https://stackoverflow.com/a/7210437

Reply:
> To my understanding "prepending" refers to the struct as it is layed out in memory. Under normal circumstances this struct will be properly packed and there will be no padding between fields. (a good goto for this topic is http://www.catb.org/esr/structure-packing/).
> This should not be an issue in this case, as there are only 32-bit ints (and the variable array at the end) in this struct. but something to keep in mind.

I'm sorry I don't get what you're looking for. Do you want to change any wording?

> 
> The other comment was mainly that to my experience "::" is very uncommon for C code documentation (it is used in C++ for namespacing) . Direct struct member access is done by "." and thus I prefer to keep this scheme also in the documentation _unless_ you are sticking to Doxygen, see https://stackoverflow.com/a/7210437

So it's a matter of taste. Since we're already using the "::" style in OP-TEE at a few places we're sticking with that.

Reply:
1. Yes, this is really a minor point that already led to too much discussion. I'd go for sth. like "[...] arrays are wrapped with this header structure", but if you prefer the existing wording, I'm fine with that too.

2. Ok. Seems I still haven't dived deep enough into the OP-TEE sources. :)